WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -210,6 +216,11 @@ final class ProductFormViewController<ViewModel: ProductFormViewModelProtocol>:
     @objc func publishProduct() {
         if viewModel.formType == .add {
             ServiceLocator.analytics.track(.addProductPublishTapped, withProperties: ["product_type": product.productType.rawValue])
+        } else if viewModel.formType == .edit && isAIContent {
```

Comment:
❓I am trying to understand why we must check for `viewModel.formType == .edit && isAIContent`. Will `isAIContent` be true in a scenario when the form type is `.add`?

Reply:
In a previous PR I had to send the form type `.edit` because otherwise the Publish button would not be visible since the product status is `draft`.

Reply:
Oh OK. Thanks for the explanation.

- We already saved the product as a draft in the AI preview screen. 
- When we navigate to the product form the form type is `edit` and not `add`. (As we are editing a `draft` product and not creating a new product)

Am I understanding this correctly?

Reply:
Yes, that's correct. I don't have the context why this event is tracked only for creating a new product and not when publishing a draft one though, so I added the check to only track for the AI case.

Reply:
Thank you for the details! 👍